sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/implementation/http/policy/ExponentialBackoffDelay.java

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.generic.core.implementation.http.policy;
+
+import com.generic.core.http.policy.RetryPolicy;
+import com.generic.core.util.configuration.Configuration;
+import com.generic.core.util.ClientLogger;
+
+import java.time.Duration;
+import java.util.Objects;
+import java.util.concurrent.ThreadLocalRandom;
+
+/**
+ * A truncated exponential backoff implementation of {@link RetryPolicy.RetryStrategy} that has a delay duration that exponentially
+ * increases with each retry attempt until an upper bound is reached after which every retry attempt is delayed by the
+ * provided max delay duration.
+ */
+public class ExponentialBackoffDelay implements RetryPolicy.RetryStrategy {
+    private static final double JITTER_FACTOR = 0.05;
+    private static final Duration DEFAULT_BASE_DELAY = Duration.ofMillis(800);
+    private static final Duration DEFAULT_MAX_DELAY = Duration.ofSeconds(8);
+    private static final ClientLogger LOGGER = new ClientLogger(ExponentialBackoffDelay.class);
+    private final long baseDelayNanos;
+    private final long maxDelayNanos;
```

Comment:
I recently made a PR in azure-core which adds two more properties to the retry strategy implementations, `Predicate<HttpResponse>` and `Predicate<Throwable>` to allow the behavior of `shouldRetry` and `shouldRetryException` to be configurable without needed to do

```java
new ExponentialBackoffDelay() {
  @Override
  public boolean shouldRetry(HttpResponse response) {
    // do stuff
  }

  @Override
  public boolean shouldRetryException(Throwable throwable) {
    // do stuff
  }
}
```

Reply:
I think to support this we could extract `shoulRetry` APIs into a seperate configurable option, something like `configureRetryCondition()` in which the users can independently specify retry conditionals without having to go through  the Strategy or RetryOptions at all.